scene/main/window.cpp

```diff
@@ -1383,6 +1384,21 @@ void Window::_window_input(const Ref<InputEvent> &p_ev) {
 		emit_signal(SceneStringNames::get_singleton()->window_input, p_ev);
 	}
 
+	if (!Engine::get_singleton()->is_editor_hint() && p_ev->is_action_pressed("ui_take_screenshot")) {
+		// The editor already has its own screenshot taking functionality.
+		const Ref<ViewportTexture> texture = get_texture();
+		ERR_FAIL_COND_MSG(texture.is_null(), "Cannot get a viewport texture from the window for taking a screenshot.");
+		const Ref<Image> img = texture->get_image();
+		ERR_FAIL_COND_MSG(img.is_null(), "Cannot get an image from a viewport texture of the window for taking a screenshot.");
+		const String path = ProjectSettings::get_singleton()->globalize_path("user://screenshot_" + Time::get_singleton()->get_datetime_string_from_system().replace(":", "") + ".png");
+		const Error error = img->save_png(path);
+
+		ERR_FAIL_COND_MSG(error != OK, vformat("Cannot save screenshot to: %s", path));
+		// Using formatting around the printed path allows using Ctrl + Click in VS Code's integrated terminal to open the screenshot:
+		// <https://github.com/microsoft/vscode/issues/97941>
+		print_line_rich(vformat("Screenshot saved to: [u]%s[/u]", path));
```

Comment:
We should probably use this in other parts of the engine :slightly_smiling_face:
See https://github.com/microsoft/vscode/issues/97941.

Note that quoting paths will not work, so formatting must be used for "generic" paths. There are other types of paths with spaces supported, but these are intended to be used for programming stack traces.